snuba/subscriptions/scheduler_consumer.py

```diff
@@ -248,7 +249,9 @@ def __init__(
 
         self.__mode = mode
 
-        self.__partitions = stream_loader.get_default_topic_spec().partitions_number
+        self.__partitions = get_partition_count(
```

Comment:
If this is only set at the start of the scheduler, it means that the scheduler will not be able to withstand an increase in the number of partitions, it will receive a tick for an unknown partition and crash. 
I think it would be ok, but Could you  find the places where we validate that the tick received is for a known partition and provide a clear error message that explains that the crash would be normal if due to an increase in partitions ?
The goal is to reduce the noise and the complexity of the partition resize process.